internal/reconcilers/messages/messages.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2024 Stacklok, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package messages contains messages structs and builders for events
+// handled by reconcilers.
+package messages
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/ThreeDotsLabs/watermill/message"
+	"github.com/google/uuid"
+)
+
+// RepoReconcilerEvent is an event that is sent to the reconciler topic
+type RepoReconcilerEvent struct {
+	// Project is the project that the event is relevant to
+	Project uuid.UUID `json:"project"`
+	// Repository is the repository to be reconciled
+	Repository int64 `json:"repository" validate:"gte=0"`
+}
+
+// NewRepoReconcilerMessage creates a new repos init event
+func NewRepoReconcilerMessage(providerID uuid.UUID, repoID int64, projectID uuid.UUID) (*message.Message, error) {
+	evt := &RepoReconcilerEvent{
+		Repository: repoID,
+		Project:    projectID,
+	}
+
+	evtStr, err := json.Marshal(evt)
+	if err != nil {
+		return nil, fmt.Errorf("error marshalling init event: %w", err)
+	}
+
+	msg := message.NewMessage(uuid.New().String(), evtStr)
+	msg.Metadata.Set("provider_id", providerID.String())
+	return msg, nil
+}
```

Comment:
This was moved here to solve a circular dependency between `internal/reconcilers/reconcilers.go` and `internal/service/service.go`